mbusd 0.5.2 install failure

### Issue related to

Sunsynk / mbusd Home Assistant Add-On for 0.5.2

### Describe the issue/bug

When installing 0.5.2 it fails with errors surrounding the below. this was working until this version came out :


**Expected behavior**
standard add-on installations completion

### Your environment

- Home Assistant version: 2025.10.2
- Addon:
  - Name: sunsynk-multi
  - Version: 2025-01-11
- Inverter:
  - Make: Sunsynk
  - Model: EESW-D205
  - Firmware: 4790123B24R
- Adaptor details:
  - RS485

### You configuration

```

not config related, install related

```

### Logs

```Failed to install add-on
Docker build failed for bcf3623f/aarch64-addon-hass-addon-mbusd:0.5.2 (exit code 1). Build output: #0 building with "default" instance using docker driver #1 [internal] load build definition from Dockerfile #1 transferring dockerfile: 434B done #1 WARN: InvalidDefaultArgInFrom: Default value for ARG ${BUILD_FROM} results in empty or invalid base image name (line 2) #1 DONE 0.1s #2 [internal] load metadata for ghcr.io/home-assistant/aarch64-base:latest #2 DONE 0.5s #3 [internal] load .dockerignore #3 transferring context: 2B done #3 DONE 0.1s #4 [1/4] FROM ghcr.io/home-assistant/aarch64-base:latest@sha256:2bd17d24817ed7c28750e414f6d80ea65e213ee15c9b7570239400ea4d24ab3f #4 CACHED #5 [internal] load build context #5 transferring context: 608B done #5 DONE 0.1s #6 [2/4] RUN apk --no-cache add --virtual .build-deps make cmake gcc g++ git pkgconf && git clone --depth 1 --branch v0.5.2 https://github.com/3cky/mbusd.git /usr/src && cd /usr/src && ls && cmake . && make && apk del .build-deps #6 0.605 fetch https://dl-cdn.alpinelinux.org/alpine/v3.22/main/aarch64/APKINDEX.tar.gz #6 2.098 fetch https://dl-cdn.alpinelinux.org/alpine/v3.22/community/aarch64/APKINDEX.tar.gz #6 4.284 (1/25) Installing make (4.4.1-r3) #6 4.483 (2/25) Installing acl-libs (2.3.2-r1) #6 4.702 (3/25) Installing libbz2 (1.0.8-r6) #6 4.883 (4/25) Installing libexpat (2.7.3-r0) #6 5.071 (5/25) Installing lz4-libs (1.10.0-r0) #6 5.258 (6/25) Installing libarchive (3.8.1-r0) #6 5.468 (7/25) Installing rhash-libs (1.4.5-r0) #6 5.658 (8/25) Installing cmake (3.31.7-r1) #6 8.975 (9/25) Installing jansson (2.14.1-r0) #6 9.160 (10/25) Installing binutils (2.44-r3) #6 9.841 (11/25) Installing libgomp (14.2.0-r6) #6 10.03 (12/25) Installing libatomic (14.2.0-r6) #6 10.21 (13/25) Installing gmp (6.3.0-r3) #6 10.41 (14/25) Installing isl26 (0.26-r1) #6 10.67 (15/25) Installing mpfr4 (4.2.1_p1-r0) #6 10.87 (16/25) Installing mpc1 (1.3.1-r1) #6 11.05 (17/25) Installing gcc (14.2.0-r6) #6 15.39 (18/25) Installing libstdc++-dev (14.2.0-r6) #6 16.17 (19/25) Installing musl-dev (1.2.5-r10) #6 16.63 (20/25) Installing g++ (14.2.0-r6) #6 17.88 (21/25) Installing pcre2 (10.43-r1) #6 18.08 (22/25) Installing git (2.49.1-r0) #6 18.50 (23/25) Installing git-init-template (2.49.1-r0) #6 18.67 (24/25) Installing pkgconf (2.4.3-r0) #6 18.86 (25/25) Installing .build-deps (20251017.072049) #6 18.86 Executing busybox-1.37.0-r19.trigger #6 19.05 OK: 294 MiB in 75 packages #6 19.30 Cloning into '/usr/src'... #6 20.39 warning: refs/tags/v0.5.2 cd5f20a35ad626c6562202ee12238cf496e66848 is not a commit! #6 20.39 Note: switching to '91f8e35eae45feee32f74a3201a772f863d14aca'. #6 20.39 #6 20.39 You are in 'detached HEAD' state. You can look around, make experimental #6 20.39 changes and commit them, and you can discard any commits you make in this #6 20.39 state without impacting any branches by switching back to a branch. #6 20.39 #6 20.39 If you want to create a new branch to retain commits you create, you may #6 20.39 do so (now or later) by using -c with the switch command. Example: #6 20.39 #6 20.39 git switch -c <new-branch-name> #6 20.39 #6 20.39 Or undo this operation with: #6 20.39 #6 20.39 git switch - #6 20.39 #6 20.39 Turn off this advice by setting config variable advice.detachedHead to false #6 20.39 #6 20.42 CHANGELOG.md #6 20.42 CMakeLists.txt #6 20.42 Dockerfile #6 20.42 LICENSE #6 20.42 README.md #6 20.42 cmake #6 20.42 conf #6 20.42 doc #6 20.42 src #6 20.42 systemd-units #6 20.42 tests #6 20.44 CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required): #6 20.44 Compatibility with CMake < 3.10 will be removed from a future version of #6 20.44 CMake. #6 20.44 #6 20.44 Update the VERSION argument <min> value. Or, use the <min>...<max> syntax #6 20.44 to tell CMake that the project requires at least <min> but has been updated #6 20.44 to work with policies introduced by <max> or earlier. #6 20.44 #6 20.44 #6 20.64 -- The C compiler identification is GNU 14.2.0 #6 20.84 -- The CXX compiler identification is GNU 14.2.0 #6 20.87 -- Detecting C compiler ABI info #6 21.10 -- Detecting C compiler ABI info - done #6 21.17 -- Check for working C compiler: /usr/bin/cc - skipped #6 21.17 -- Detecting C compile features #6 21.17 -- Detecting C compile features - done #6 21.21 -- Detecting CXX compiler ABI info #6 21.47 -- Detecting CXX compiler ABI info - done #6 21.54 -- Check for working CXX compiler: /usr/bin/c++ - skipped #6 21.54 -- Detecting CXX compile features #6 21.55 -- Detecting CXX compile features - done #6 21.59 -- Found UnixCommands: /bin/bash #6 21.61 -- Checking for module 'systemd' #6 21.62 -- Package 'systemd' not found #6 21.62 -- Looking for cfmakeraw #6 21.85 -- Looking for cfmakeraw - found #6 21.85 -- Looking for cfsetspeed #6 22.07 -- Looking for cfsetspeed - found #6 22.08 -- Looking for cfsetispeed #6 22.30 -- Looking for cfsetispeed - found #6 22.30 -- Looking for TIOCSRS485 #6 22.54 -- Looking for TIOCSRS485 - found #6 22.54 -- Looking for time #6 22.78 -- Looking for time - found #6 22.78 -- Looking for localtime #6 23.03 -- Looking for localtime - found #6 23.03 -- Passing HRDATE to compiler space #6 23.03 -- Looking for tty_get_name in util #6 23.26 -- Looking for tty_get_name in util - not found #6 23.26 -- Looking for uu_lock in util #6 23.49 -- Looking for uu_lock in util - not found #6 23.56 -- Install prefix /usr/local #6 23.56 -- Install bindir: /usr/local/bin #6 23.56 -- Install sysconfdir: /usr/local/etc #6 23.56 -- Install datadir: /usr/local/share #6 23.56 -- Configuring done (3.1s) #6 23.58 -- Generating done (0.0s) #6 23.58 -- Build files have been written to: /usr/src #6 23.68 [ 8%] Building C object CMakeFiles/mbusd.dir/src/main.c.o #6 23.78 In file included from /usr/src/src/cfg.h:39, #6 23.78 from /usr/src/src/main.c:37: #6 23.78 /usr/src/src/tty.h:38:10: fatal error: linux/serial.h: No such file or directory #6 23.78 38 | #include <linux/serial.h> #6 23.78 | ^~~~~~~~~~~~~~~~ #6 23.78 compilation terminated. #6 23.79 make[2]: *** [CMakeFiles/mbusd.dir/build.make:79: CMakeFiles/mbusd.dir/src/main.c.o] Error 1 #6 23.79 make[1]: *** [CMakeFiles/Makefile2:88: CMakeFiles/mbusd.dir/all] Error 2 #6 23.79 make: *** [Makefile:166: all] Error 2 #6 ERROR: process "/bin/ash -o pipefail -c apk --no-cache add --virtual .build-deps make cmake gcc g++ git pkgconf && git clone --depth 1 --branch v0.5.2 https://github.com/3cky/mbusd.git /usr/src && cd /usr/src && ls && cmake . && make && apk del .build-deps" did not complete successfully: exit code: 2 ------ > [2/4] RUN apk --no-cache add --virtual .build-deps make cmake gcc g++ git pkgconf && git clone --depth 1 --branch v0.5.2 https://github.com/3cky/mbusd.git /usr/src && cd /usr/src && ls && cmake . && make && apk del .build-deps: 23.68 [ 8%] Building C object CMakeFiles/mbusd.dir/src/main.c.o 23.78 In file included from /usr/src/src/cfg.h:39, 23.78 from /usr/src/src/main.c:37: 23.78 /usr/src/src/tty.h:38:10: fatal error: linux/serial.h: No such file or directory 23.78 38 | #include <linux/serial.h> 23.78 | ^~~~~~~~~~~~~~~~ 23.78 compilation terminated. 23.79 make[2]: *** [CMakeFiles/mbusd.dir/build.make:79: CMakeFiles/mbusd.dir/src/main.c.o] Error 1 23.79 make[1]: *** [CMakeFiles/Makefile2:88: CMakeFiles/mbusd.dir/all] Error 2 23.79 make: *** [Makefile:166: all] Error 2 ------ [33m1 warning found (use docker --debug to expand): [0m - InvalidDefaultArgInFrom: Default value for ARG ${BUILD_FROM} results in empty or invalid base image name (line 2) Dockerfile:4 -------------------- 3 | 4 | >>> RUN \ 5 | >>> apk --no-cache add --virtual .build-deps \ 6 | >>> make cmake gcc g++ git pkgconf \ 7 | >>> && git clone --depth 1 --branch v0.5.2 \ 8 | >>> https://github.com/3cky/mbusd.git /usr/src \ 9 | >>> && cd /usr/src \ 10 | >>> && ls \ 11 | >>> && cmake . \ 12 | >>> && make \ 13 | >>> && apk del .build-deps 14 | -------------------- ERROR: failed to build: failed to solve: process "/bin/ash -o pipefail -c apk --no-cache add --virtual .build-deps make cmake gcc g++ git pkgconf && git clone --depth 1 --branch v0.5.2 https://github.com/3cky/mbusd.git /usr/src && cd /usr/src && ls && cmake . && make && apk del .build-deps" did not complete successfully: exit code: 2
```

--- a/hass-addon-mbusd/Dockerfile
+++ b/hass-addon-mbusd/Dockerfile
@@ -1,19 +1,19 @@
-ARG BUILD_FROM
-FROM ${BUILD_FROM}
+ARG BUILD_FROM=ghcr.io/home-assistant/aarch64-base
 
-RUN \
-    apk --no-cache add  --virtual .build-deps \
-        make cmake gcc g++ git pkgconf \
-    && git clone --depth 1 --branch v0.5.2 \
-        https://github.com/3cky/mbusd.git /usr/src \
-    && cd /usr/src \
-    && ls \
-    && cmake . \
-    && make \
-    && apk del .build-deps
-
+# mbusd Dockerfile as reference https://github.com/3cky/mbusd/blob/master/Dockerfile
+# - FROM home-assistant/base iso alpine:latest
+# - Clone a specific git tag from https://github.com/3cky/mbusd/releases
+FROM ${BUILD_FROM} AS build
+RUN apk add --no-cache alpine-sdk cmake linux-headers git
 WORKDIR /usr/src
+RUN git clone --depth 1 --branch v0.5.2 https://github.com/3cky/mbusd.git
+RUN cd mbusd \
+    && cmake -DCMAKE_INSTALL_PREFIX=/usr . \
+    && make && make install
 
+FROM ${BUILD_FROM} AS runtime
+COPY --from=build /usr/bin/mbusd /usr/bin/mbusd
+WORKDIR /usr/src
 COPY run.sh run.sh
 ENTRYPOINT [ "./run.sh" ]
 

--- a/hass-addon-mbusd/build.yaml
+++ b/hass-addon-mbusd/build.yaml
@@ -1,6 +1,7 @@
+# https://github.com/home-assistant/docker-base
 build_from:
-  aarch64: "ghcr.io/home-assistant/aarch64-base"
-  armv7: "ghcr.io/home-assistant/armv7-base"
-  armhf: "ghcr.io/home-assistant/armhf-base"
-  amd64: "ghcr.io/home-assistant/amd64-base"
-  i386: "ghcr.io/home-assistant/i386-base"
+  aarch64: "ghcr.io/home-assistant/aarch64-base:3.22"
+  armv7: "ghcr.io/home-assistant/armv7-base:3.22"
+  armhf: "ghcr.io/home-assistant/armhf-base:3.22"
+  amd64: "ghcr.io/home-assistant/amd64-base:3.22"
+  i386: "ghcr.io/home-assistant/i386-base:3.22"

--- a/hass-addon-mbusd/config.yaml
+++ b/hass-addon-mbusd/config.yaml
@@ -1,11 +1,12 @@
 name: Modbus TCP to Modbus RTU Gateway Add-on
 slug: hass-addon-mbusd
 description: Add-on for a Modbus TCP to Modbus RTU Gateway using mbusd
-version: "0.5.2"
+version: "0.5.2b"
 startup: services
 boot: auto
 url: 'https://kellerza.github.io/sunsynk/guide/mbusd'
 uart: true
+apparmor: true
 arch:
   - aarch64
   - amd64
@@ -16,7 +17,7 @@ ports:
   '502': 502
 ports_description:
   '502': Modbus
-host_network: true
+# host_network: true
 options:
   DEVICE: /dev/ttyUSB0
   BAUDRATE: 9600

--- a/hass-addon-mbusd/run.sh
+++ b/hass-addon-mbusd/run.sh
@@ -22,4 +22,4 @@ fi
 
 bashio::log.info "Starting mbusd -d -L - -v $LOGLEVEL -p $DEVICE -s $BAUDRATE -m $MODE -P 502"
 
-exec /usr/src/mbusd -d -L - -v $LOGLEVEL -p $DEVICE -s $BAUDRATE -m $MODE -P 502 -T $TIMEOUT
+exec /usr/bin/mbusd -d -L - -v $LOGLEVEL -p $DEVICE -s $BAUDRATE -m $MODE -P 502 -T $TIMEOUT

--- a/scripts/copy2local.cmd
+++ b/scripts/copy2local.cmd
@@ -1,9 +1,17 @@
 @echo off
-CALL :copy2 hass-addon-sunsynk-multi,\\192.168.1.8\addons\hass-addon-sunsynk-multi\
+@rem CALL :copy_ss hass-addon-sunsynk-multi,\\192.168.1.8\addons\hass-addon-sunsynk-multi
+CALL :copy_addon hass-addon-mbusd,\\192.168.1.8\addons\hass-addon-mbusd
 
 EXIT /B %ERRORLEVEL%
 
-:copy2
+:copy_ss
+
+echo # Copy sunsynk package
+for %%f in (pyproject.toml,MANIFEST.in,LICENSE,README.md,uv.lock) do xcopy /Y "%%f" %~2\sunsynk\
+xcopy /Y /S /EXCLUDE:scripts\copyexclude.txt src %~2\sunsynk\src\
+
+
+:copy_addon
 echo # Modify Config for local testing
 set cf=%~1\config.localtest.yaml
 cp %~1\config.yaml %cf%
@@ -13,9 +21,5 @@ xcopy /Y %cf% %~2\config.yaml
 
 echo # Copy '%~1' to '%~2'
 xcopy /Y /S /EXCLUDE:scripts\copyexclude.txt %~1 %~2
-
-echo # Copy sunsynk package
-for %%f in (pyproject.toml,MANIFEST.in,LICENSE,README.md,uv.lock) do xcopy /Y "%%f" %~2\sunsynk\
-xcopy /Y /S /EXCLUDE:scripts\copyexclude.txt src %~2\sunsynk\src\
 
 EXIT /B 0

